src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -857,6 +875,36 @@ else if (shouldBootstrap())
         }
     }
 
+    public void waitForSchema(int delay)
+    {
+        // first sleep the delay to make sure we see all our peers
+        for (long i = 0; i < delay; i += 1000)
+        {
+            // if we see schema, we can proceed to the next check directly
+            if (!Schema.instance.isEmpty())
```

Comment:
Thrift and native connections are only enabled on a node once the node finishes joining, correct? Or more generally, do we know for sure that in all cases, when we start a new node, the only existing keyspace will be the system keyspace (and therefore schemaVersion will match empty schema)?